src/main.js

```diff
@@ -299,5 +299,25 @@ const parse = envSpecTxt => {
   });
 };
 
+/** @function serializeForm
+ * @desc transforms a HTML form to an array of variables with their values if given by user
+ * @param {HTML form} form the form completed by the user
+ * @returns {promise} that resolves to an array of strings
+ */
+const serializeForm = form => {
+  outputForm = [];
+  newEntryForm = "NOTHING IS DONE";
+  form.querySelectorAll("select,input").forEach(input => {
+    newEntryForm = input.value
```

Comment:
Instead of returning a `string` we want to return an object. Also you can simplify this:

```js
outputForm.push({
  input.name.toUpperCase(): input.value || null
})
```